docs/source/development/widget.md

```diff
@@ -251,9 +250,101 @@ See [Storybook](https://6.docs.plone.org/storybook) with available widgets.
 
 ## Write a new widget
 
-```{note}
-Please contribute to this section!
-```
+A custom widget is often used in Volto forms.
+To configure a new Volto widget follow these steps.
+
+1.  Add the following import as the first line in the file {file}`index.js` in the add-on to import the component of the actual widget.
+
+    ```jsx
+    import { ToggleWidget } from './ToggleWidget';
+    ```
+
+1.  Add the following line before returning from the `applyConfig` function inside the file {file}`index.js` in the add-on to register the widget add-on component.
+
+    ```jsx
+    config.widgets.widget.toggle_widget = ToggleWidget;
+    ```
+
+1.  Next, create the widget component file {file}`ToggleWidget.jsx` with the following content.
+
+    ```jsx
+    import { Button } from "semantic-ui-react";
+    export function ToggleWidget({ id, value, onChange }) {
+      return (
+        <Button
+          onClick={(e) => {
+            e.preventDefault();
+            onChange(id, value === "On" ? "Off" : "On");
+          }}
+        >
+          Toggle me: {value}
+        </Button>
+      );
+    }
+
+    export default ToggleWidget;
+    ```
+
+1.  Install `semantic-ui-react` as a dependency of your add-on to make it work.
+
+    ```shell
+    npm i semantic-ui-react
+    ```
+
+1.  Using the component shadowing technique, create a directory in your add-on {file}`src/customizations/components/manage/Sidebar/`.
+     Copy the file in Volto core {file}`components/manage/Sidebar/Sidebar.jsx` to the directory you just created.
+     This file shadows the `Sidebar` component inside the corresponding core Volto file.
+
+1.  Inside the new {file}`src/customizations/components/manage/Sidebar/Sidebar.jsx`, insert the following import at the top.
+
+    ```jsx
+    import { Form } from '@plone/volto/components/manage/Form';
+    ```
+
+1.  Below the imports, write the following example custom form, which has one field of type `toggle_widget` to serve as our widget.
+
+    ```jsx
+    const MyForm = () => {
+      return (
+        <Form
+          schema={{
+            fieldsets: [
+              {
+                id: "default",
+                title: "Title",
+                fields: ["toggle"],
+              },
+            ],
+            properties: {
+              toggle: {
+                title: "My toggle",
+                widget: "toggle_widget",
+                default: "On",
+              },
+            },
+            required: [],
+          }}
+        />
+      );
+    };
+    ```
+
+1.  To see the widget inside the `Sidebar` component on the right side of the Volto UI, add the following line inside the `(React.)Fragment` tag that is rendered.
+
+    ```jsx
+    <MyForm />
+    ```
+
+1.  Finally, restart Volto by pressing Ctrl-C inside the shell instance which runs it, then restart it with the following command.
```

Comment:
We have references for MyST markup, listed in order of which to check. Plone's reference is limited to the most frequently used markup, then MyST which has most roles and directives, and ultimately Sphinx. Unfortunately this is one of those that is only documented in the grandparent in Sphinx's documentation.

- https://6.docs.plone.org/contributing/documentation/myst-reference.html
- https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#roles-an-in-line-extension-point
- https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-kbd

```suggestion
1.  Finally, restart Volto by pressing {kbd}`Ctrl-C` inside the active frontend shell session, then restart it with the following command.
```